magda-web-client/src/Components/Dataset/Add/AddDatasetLinkSection.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import {
+    File,
+    FileState,
+    FileSource
+} from "Components/Dataset/Add/DatasetAddCommon";
+import "./AddDatasetLinkSection.scss";
+
+import isUrl from "is-url";
+import uuid from "uuid";
+import DatasetLinkItem from "./DatasetLinkItem";
+
+type Props = {
+    files: File[];
+    addFile: (file: File) => void;
+    editFile: (index: number) => (updater: (file: File) => File) => void;
+};
+
+const AddDatasetLinkSection = (props: Props) => {
+    const [url, setUrl] = useState("");
+    const [validationErrorMessage, setValidationErrorMessage] = useState("");
+    const files = props.files
+        .map((item, idx) => ({ file: item, idx }))
+        .filter(item => item.file.creationSource === FileSource.DatasetUrl);
+
+    const fetchUrl = () => {
+        if (!isUrl(url)) {
+            setValidationErrorMessage("Please input an valid URL!");
+        } else {
+            setValidationErrorMessage("");
+
+            props.addFile({
+                id: uuid.v4(),
+                downloadURL: url,
+                creationSource: FileSource.DatasetUrl,
+                title: url,
+                modified: new Date(),
+                format: "",
+                _state: FileState.Processing,
+                _progress: 50
+            });
+
+            setUrl("");
```

Comment:
This isn't working for some reason - the url box still has the URL in it after you hit fetch:
<img width="892" alt="image" src="https://user-images.githubusercontent.com/900555/74990995-e6132d00-5498-11ea-8d98-b2f1cc1a6129.png">

Reply:
It's strange... Checked the preview site in the cluster and it seems the line is not there:
![image](https://user-images.githubusercontent.com/674387/74993027-e4983380-549d-11ea-9e57-2e9738eb0cfc.png)
Probably gitlab didn't checkout correct commit?
Will look into it~

Reply:
Yeah~ it seems a racing condition of the CI pipeline~
I re-run the UI docker build step, re-deploy the UI preview and it seems works ok now~